pkg/vm/engine/disttae/db.go

```diff
@@ -19,20 +19,52 @@ import (
 
 	"github.com/matrixorigin/matrixone/pkg/pb/plan"
 	"github.com/matrixorigin/matrixone/pkg/pb/timestamp"
+	"github.com/matrixorigin/matrixone/pkg/txn/client"
 	"github.com/matrixorigin/matrixone/pkg/vm/engine"
 )
 
-func newDB() *DB {
-	return &DB{
-		readTs: timestamp.Timestamp{
-			PhysicalTime: 0,
-			LogicalTime:  0,
-		},
+func newDB(cli client.TxnClient, dnList []DNStore) *DB {
+	dnMap := make(map[string]int)
+	for i := range dnList {
+		dnMap[dnList[i].UUID] = i
 	}
+	db := &DB{
+		cli:    cli,
+		dnMap:  dnMap,
+		tables: make(map[[2]uint64]Partitions),
+	}
+	return db
 }
 
 func (db *DB) Update(ctx context.Context, dnList []DNStore,
 	databaseId, tableId uint64, ts timestamp.Timestamp) error {
+	op, err := db.cli.New()
+	if err != nil {
+		return err
+	}
+	db.Lock()
```

Comment:
get an anonymous function and call defer?

Reply:
yes